packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js

```diff
@@ -262,6 +263,12 @@ function verifyTypeScriptSetup() {
     );
   }
 
+  if (parsedTsConfig.paths != null && parsedTsConfig.baseUrl == null) {
+    messages.push(
+      `${chalk.cyan('paths')} requires ${chalk.cyan('baseUrl')} to be set`
+    );
+  }
+
```

Comment:
I'm not sure if this is enough. I think this should probably be an error and cause everything to stop.